compiler/modules/types/flaot.bal

```diff
@@ -0,0 +1,98 @@
+// Implementation specific to basic type float.
+
+type FloatSubtype readonly & record {|
+    boolean allowed;
+    float[] values;
+|};
+
+public function floatConst(float value) returns SemType {
+    FloatSubtype st = { allowed: true, values: [value] };
+    return uniformSubtype(UT_FLOAT, st);
+}
+
+function floatSubtypeUnion(SubtypeData d1, SubtypeData d2) returns SubtypeData {
+    FloatSubtype t1 = <FloatSubtype>d1;
+    FloatSubtype t2 = <FloatSubtype>d2;
+    FloatSubtype res;
+    boolean b1 = t1.allowed;
+    boolean b2 = t2.allowed;
+    if b1 && b2 {
+       return createFloatSubtype(true, floatListUnion(t1.values, t2.values));
+    }
+    else if !b1 && !b2 {
+       return createFloatSubtype(false, floatListIntersect(t1.values, t2.values));
+    }
+    else if b1 && !b2 {
+       return createFloatSubtype(false, floatListDiff(t2.values, t1.values));
+    }
+    else {
+        // !b1 && b2
+       return createFloatSubtype(false, floatListDiff(t1.values, t2.values));
+    }
+}
+
+function floatSubtypeIntersect(SubtypeData d1, SubtypeData d2) returns SubtypeData {
+    FloatSubtype t1 = <FloatSubtype>d1;
```

Comment:
Should I generify this part by creating `type ListableSubtype FloatSubtype|StringSubtype`

Reply:
Yes, `EnumerableSubtype`